common/constants.go

```diff
@@ -167,3 +173,15 @@ const (
 	EqClassEquipment EquipmentClass = 5
 	EqClassGrenade   EquipmentClass = 6
 )
+
+const (
+	ServerClassBaseGrenadeProjectile ServerClassID = 9
+	ServerClassDecoyProjectile       ServerClassID = 41
+	ServerClassMolotovProjectile     ServerClassID = 98
+	ServerClassSmokeProjectile       ServerClassID = 134
+)
+
+const (
+	EqModelFlash EquipmentModelIndex = 131
```

Comment:
These values are not constant between demos. Maybe get filenames from model index to figure out which nade.

If we'll be doing this for BaseGrenadeProjectile, I think we might as well do the same for {Decoy, Molotov, Smoke}Projectile. This is for backwards compatibility as these classes can have different values in older demos. E.g. I know that MolotovProjectile had class ID 97 previously.

Reply:
Yeah, I thought that it might be the case that they change between demos. Do you happen to know where we can read this information?

Reply:
It is in the modelprecache. The model index can be used on the modelPreCache (on Parser) to get the relevant model path. I am guessing doing this once and persisting the mapping would be best?

So e.g. something like this pseudo code:

- Expose modelPreCache

```
func (p *Parser) ModelPreCache() []string {
	return p.modelPreCache
}
```

- Set up handlers to get the precache when it has been populated

```
p.RegisterEventHandler(func(e events.StringTableCreatedEvent) {
	if e.TableName == "modelprecache" {
		// Now we can find the model index for each nade based on name
		FindMdlIdx(p.ModelPreCache())
	}
})
```

- Find the index for the different nade projectiles, and then persist it somewhere?

```
func FindMdlIdx(m []string) {
	for i, n in := range m {
		if strings.Contains(m, "flashbang_dropped") {
			// i is the index of flashbang projectile
			// ... and so forth for every nade
		}
	}
}
```

Reply:
That was an excellent description, thank you! I've given the implementation an attempt.

How did you know this? I'm pretty new to parsing these demo files, and I've spent a lot of time trying to find _any_ documentation on them. I've only had luck with stumbling over info on random forums, and various other implementations of this library.